### PR TITLE
feat: Add requirements.txt files and update gcloud version

### PR DESCRIPTION
**Pull request by chalmerlowe**
Updated three other python docker images to use `requirements.txt` with hashes.
Also updated the `gcloud CLI` version in one of the Dockerfiles to a more modern version.
Adds `mamba` to one of the Dockerfiles (python-multi) to enable testing of installs within a conda/mamba virtual environment.

### CODE REVIEW

File: python/cloud-devrel-kokoro-resources/google-cloud-python/autosynth/Dockerfile [hunk 1]
```diff
@@ -20,5 +20,7 @@ ENTRYPOINT /bin/bash
 RUN pyenv install 3.7.2
 RUN pyenv global 3.7.2
 
-RUN python3 -m pip install nox
+COPY requirements.txt /requirements.txt
+RUN python -m pip install --user --require-hashes --no-cache-dir -r /requirements.txt
```
Comment (chalmerlowe) [inline python/cloud-devrel-kokoro-resources/google-cloud-python/autosynth/Dockerfile:24]:
```suggestion
RUN python -m pip install --require-hashes --no-cache-dir -r /requirements.txt
```

**Comment by chalmerlowe** [comment]
As best I can tell, something is not quite right with the hashes files. Will dig into that and see why they are not capturing all the reqs and associated hashes.

File: python/cloud-devrel-kokoro-resources/python-base/Dockerfile [hunk 2]
```diff
@@ -118,11 +118,13 @@ RUN for PYTHON_VERSION in 2.7.18 3.6.15; do \
   && rm -r "$GNUPGHOME"
 
 # Make sure the Python 3 version of pip is up-to-date and install virtualenv
+COPY requirements.txt /requirements.txt
+
 RUN /usr/local/bin/python3 -m pip \
   install \
   --no-cache-dir \
   --upgrade \
```
Comment (chalmerlowe) [inline python/cloud-devrel-kokoro-resources/python-base/Dockerfile:126]:
```suggestion
  --upgrade pip \
```

File: python/cloud-devrel-kokoro-resources/python-base/Dockerfile [hunk 3]
```diff
@@ -118,11 +118,13 @@ RUN for PYTHON_VERSION in 2.7.18 3.6.15; do \
...
 # Make sure the Python 3 version of pip is up-to-date and install virtualenv
+COPY requirements.txt /requirements.txt
+
 RUN /usr/local/bin/python3 -m pip \
   install \
   --no-cache-dir \
-  --upgrade \
-  pip \
-  virtualenv
+  --upgrade pip \
```
Comment (chalmerlowe) [inline python/cloud-devrel-kokoro-resources/python-base/Dockerfile:126]:
```suggestion
```

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**